optimum/intel/openvino/trainer.py

```diff
@@ -0,0 +1,558 @@
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import inspect
+import io
+import math
+import os
+import sys
+import time
+import warnings
+from itertools import chain
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional, Tuple, Union
+
+import torch
+from packaging import version
+from torch.onnx import export as onnx_export
+from torch.utils.data import DataLoader, Dataset, RandomSampler
+from torch.utils.data.distributed import DistributedSampler
+from tqdm.auto import tqdm
+from transformers import Trainer
+from transformers.data.data_collator import DataCollator
+from transformers.debug_utils import DebugOption, DebugUnderflowOverflow
+from transformers.integrations import hp_params
+from transformers.modeling_utils import PreTrainedModel, unwrap_model
+from transformers.onnx import FeaturesManager, OnnxConfig
+from transformers.tokenization_utils_base import PreTrainedTokenizerBase
+from transformers.trainer import TRAINER_STATE_NAME, TRAINING_ARGS_NAME
+from transformers.trainer_callback import TrainerCallback, TrainerState
+from transformers.trainer_pt_utils import IterableDatasetShard
+from transformers.trainer_utils import EvalPrediction, HPSearchBackend, TrainOutput, has_length, speed_metrics
+from transformers.training_args import TrainingArguments
+from transformers.utils import WEIGHTS_NAME, TensorType, is_sagemaker_mp_enabled, logging
+
+import openvino
+from nncf import NNCFConfig
+from nncf.config.structures import BNAdaptationInitArgs, QuantizationRangeInitArgs
+from nncf.torch import create_compressed_model
+from nncf.torch.nncf_network import NNCFNetwork
+from openvino.offline_transformations import compress_quantize_weights_transformation
+from openvino.runtime import Core
+from optimum.utils import logging
+
+from .configuration import OVConfig
+from .quantization import OVDataLoader
+from .utils import OV_XML_FILE_NAME
+
+
+core = Core()
+
+logger = logging.get_logger(__name__)
+
+
+class OVTrainer(Trainer):
+    """
+    OVTrainer enables NNCF quantization aware training.
+    """
+
+    def __init__(
+        self,
+        model: Union[PreTrainedModel, torch.nn.Module] = None,
+        args: TrainingArguments = None,
+        data_collator: Optional[DataCollator] = None,
+        train_dataset: Optional[Dataset] = None,
+        eval_dataset: Optional[Dataset] = None,
+        tokenizer: Optional[PreTrainedTokenizerBase] = None,
+        model_init: Callable[[], PreTrainedModel] = None,
+        compute_metrics: Optional[Callable[[EvalPrediction], Dict]] = None,
+        callbacks: Optional[List[TrainerCallback]] = None,
+        optimizers: Tuple[torch.optim.Optimizer, torch.optim.lr_scheduler.LambdaLR] = (None, None),
+        preprocess_logits_for_metrics: Callable[[torch.Tensor, torch.Tensor], torch.Tensor] = None,
+        ov_config: Optional[OVConfig] = None,
```

Comment:
I would call it `nncf_config`.

Reply:
The configuration could contain information more general to OpenVINO (related to the export for example), so I would be in favor of not constraining it to nncf, what do you think ?

Reply:
this was addressed in the previous PR that is merged already.